host/xtest/pkcs11_1000.c

```diff
@@ -8016,3 +8016,442 @@ static void xtest_pkcs11_test_1025(ADBG_Case_t *c)
 
 ADBG_CASE_DEFINE(pkcs11, 1025, xtest_pkcs11_test_1025,
 		 "PKCS11: EDDSA key generation and signing");
+
+#define RSA_AES_MAX_KEY_SIZE 32
+
+#define RSA_AES_WRAP_KEY(vect) \
+	{ \
+		vect ## _modulus, ARRAY_SIZE(vect ## _modulus), \
+		vect ## _pub_exp, ARRAY_SIZE(vect ## _pub_exp), \
+		vect ## _priv_exp, ARRAY_SIZE(vect ## _priv_exp), \
+		vect ## _prime1, ARRAY_SIZE(vect ## _prime1), \
+		vect ## _prime2, ARRAY_SIZE(vect ## _prime2), \
+	}
+
+#define RSA_AES_WRAP_RSA(vect) \
+	{ .rsa = RSA_AES_WRAP_KEY(vect) }
+
+#define RSA_AES_WRAP_AES(size) { .aes = { (size) } }
+
+static struct rsa_aes_wrap_test {
+	CK_KEY_TYPE target_type;
+	union {
+		struct {
+			const uint8_t *modulus;
+			size_t modulus_len;
+
+			const uint8_t *pub_exp;
+			size_t pub_exp_len;
+
+			const uint8_t *priv_exp;
+			size_t priv_exp_len;
+
+			const uint8_t *prime1;
+			size_t prime1_len;
+			const uint8_t *prime2;
+			size_t prime2_len;
+		} rsa;
+		struct {
+			CK_ULONG size;
+		} aes;
+	} target;
+
+	struct {
+		const uint8_t *modulus;
+		size_t modulus_len;
+
+		const uint8_t *pub_exp;
+		size_t pub_exp_len;
+
+		const uint8_t *priv_exp;
+		size_t priv_exp_len;
+
+		const uint8_t *prime1;
+		size_t prime1_len;
+		const uint8_t *prime2;
+		size_t prime2_len;
+	} key;
+} rsa_aes_wrap_tests[] = {
+	{ CKK_AES, RSA_AES_WRAP_AES(128), RSA_AES_WRAP_KEY(ac_rsassa_vect2) },
+	{ CKK_AES, RSA_AES_WRAP_AES(256), RSA_AES_WRAP_KEY(ac_rsassa_vect18) },
+	{ CKK_AES, RSA_AES_WRAP_AES(192), RSA_AES_WRAP_KEY(ac_rsassa_vect19) },
+	{ CKK_RSA, RSA_AES_WRAP_RSA(ac_rsassa_vect1),
+		   RSA_AES_WRAP_KEY(ac_rsassa_vect2) },
+	{ CKK_RSA, RSA_AES_WRAP_RSA(ac_rsassa_vect1),
+		   RSA_AES_WRAP_KEY(ac_rsassa_vect19) },
+	{ CKK_RSA, RSA_AES_WRAP_RSA(ac_rsassa_vect2),
+		   RSA_AES_WRAP_KEY(ac_rsassa_vect18) },
+	{ CKK_RSA, RSA_AES_WRAP_RSA(ac_rsassa_vect18),
+		   RSA_AES_WRAP_KEY(ac_rsassa_vect2) },
+	{ CKK_RSA, RSA_AES_WRAP_RSA(ac_rsassa_vect19),
+		   RSA_AES_WRAP_KEY(ac_rsassa_vect2) },
+	{ CKK_RSA, RSA_AES_WRAP_RSA(ac_rsassa_vect19),
+		   RSA_AES_WRAP_KEY(ac_rsassa_vect18) },
+	{ CKK_RSA, RSA_AES_WRAP_RSA(ac_rsassa_vect18),
+		   RSA_AES_WRAP_KEY(ac_rsassa_vect19) },
+};
+
+static CK_RV test_rsa_aes_wrap(ADBG_Case_t *c, CK_SESSION_HANDLE session,
+			       struct rsa_aes_wrap_test *t)
+{
+	CK_RV rv = CKR_GENERAL_ERROR;
+	CK_OBJECT_HANDLE wrapping_key = CK_INVALID_HANDLE;
+	CK_OBJECT_HANDLE unwrapping_key = CK_INVALID_HANDLE;
+	CK_OBJECT_HANDLE target_key = CK_INVALID_HANDLE;
+	CK_OBJECT_HANDLE unwrapped_key = CK_INVALID_HANDLE;
+	CK_ULONG target_size = t->target_type == CKK_AES ?
+						 t->target.aes.size / 8:
+						 t->target.rsa.modulus_len;
```

Comment:
indentation to fix.

```c
	CK_ULONG target_size = t->target_type == CKK_AES ?
			       t->target.aes.size / 8:
			       t->target.rsa.modulus_len;

Reply:
fixed in [a4f3f71](https://github.com/OP-TEE/optee_test/pull/627/commits/a4f3f7118946a5f2be78e74c1bc82c65445f9752)